### DISCUSSION TITLE
Is it possible to somehow subscribe to 'append' and 'remove' method from 'useFieldArray' ?

### DISCUSSION THREAD

**Discussion by oleksandrfomin5995** [react-hook-form/react-hook-form#9270] [Q&A] [answered]
I've got a dynamic list of items which when focused reveal a form related to a particular item, my goal is to focus on a new item when it's created by calling `append `(therefore I need access to an ID of a newely created item) or focus on a previous item when `remove` is called. 
At the moment I'm relying on `useEffect` and `fields.length` to manage focus state but I can't get it right since I don't know whether the item was removed or added when `useEffect ` fires. 

So my question is, is it possible to subscribes to either of those methods to call different function depending on what `useFieldArray` method was used?

**Comment by leapful** [comment, marked as answer]
@oleksandrfomin5995 

You can keep track on old length of `fields` by using React.useRef then comparing old length with new length inside React.useEffect. If old length > new length, the item is being removed, otherwise, it's added. Just remember to update old length ref at the end of useEffect to keep update old length as previous value of `fields` length.
